### PR TITLE
Supporting SNMP Hosts Configurations in Ansible

### PR DESCRIPTION
**Pull request by Sara-Touqan**
##### SUMMARY

Supporting SNMP Hosts Configurations in Ansible

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

onyx_snmp_hosts
##### ADDITIONAL INFORMATION

```paste below
This module supports snmp configurations that are related to hosts.
```

### CODE REVIEW

File: lib/ansible/modules/network/onyx/onyx_snmp_hosts.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        """
+        host_spec = dict(name=dict(required=True),
+                         enabled=dict(type='bool'),
+                         notification_type=dict(type='str', choices=['trap', 'inform']),
+                         port=dict(type='str'),
+                         version=dict(type='str', choices=['1', '2c', '3']),
+                         user_name=dict(type='str'),
+                         auth_type=dict(type='str', choices=['md5', 'sha', 'sha224', 'sha256', 'sha384', 'sha512']),
+                         privacy_type=dict(type='str', choices=['3des', 'aes-128', 'aes-192', 'aes-192-cfb', 'aes-256', 'aes-256-cfb', 'des']),
+                         privacy_password=dict(type='str'),
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_snmp_hosts.py]: add no_log=True to hide passwords

File: lib/ansible/modules/network/onyx/onyx_snmp_hosts.py [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+        cmd = "show snmp host"
+        return show_cmd(self._module, cmd, json_fmt=True, fail_on_error=False)
+
+    def load_current_config(self):
+        self._current_config = dict()
+        hosts_config = self._show_hosts_config()
+        if hosts_config[1]:
+            self._set_host_config(hosts_config[1])
+
+    def generate_commands(self):
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_snmp_hosts.py:404]: function is too complex.
try to split to smaller functions

**Review by samerd — CHANGES_REQUESTED** [review]
thanks @Sara-Touqan 
added some comments, please fix

**Review by samerd — APPROVED** [review]
Thanks @Sara-Touqan 
great job

**Comment by Qalthos** [comment]
rebuild_merge

**Pull request merged**